### PR TITLE
feat(federation): respect the `subgraph_graphql_validation` option

### PR DESCRIPTION
**Pull request by duckki**
- validate subgraph queries only when `subgraph_graphql_validation` configuration option is true.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [x] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

[ROUTER-667]: https://apollographql.atlassian.net/browse/ROUTER-667?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apollo-federation/src/operation/mod.rs [hunk 1]
```diff
@@ -3955,7 +3955,7 @@ impl TryFrom<Operation> for Valid<executable::ExecutableDocument> {
         document.fragments = fragments;
         document.operations.insert(operation);
         coerce_executable_values(value.schema.schema(), &mut document);
-        Ok(document.validate(value.schema.schema())?)
+        Ok(Valid::assume_valid(document))
```
Comment (goto-bus-stop) [inline apollo-federation/src/operation/mod.rs:3958]: We should not do `assume_valid` inside the `TryFrom` implementation. If we do this we should change the impl to `TryFrom<Operation> for executable::ExecutableDocument`.

**Review by goto-bus-stop — COMMENTED** [review]
Tbh, I have a slight preference for always validating, and returning an internal error if we produced an invalid query.

**Comment by duckki** [comment]
JS QP doesn't seem to validate subgraph queries. So, Rust QP is taking a performance penalty (didn't measure how much). Also, this validation has never failed so far over the corpus.

**Comment by duckki** [comment]
This is not urgent. We can revisit this when we work on performance.

**Comment by abernix** [comment]
Closing this to clean-up some open and aging PRs, but don't hesitate to just click the re-open button it if it's still relevant.

**Pull request closed without merging**